modules/tinymce/src/core/main/ts/delete/ImageBlockDelete.ts

```diff
@@ -13,19 +13,20 @@ import CaretPosition from '../caret/CaretPosition';
 import { isAfterImageBlock, isBeforeImageBlock } from '../caret/CaretPositionPredicates';
 import { getChildNodeAtRelativeOffset } from '../caret/CaretUtils';
 
-const deleteCaret = (editor: Editor, forward: boolean): boolean => {
+const deleteCaret = (editor: Editor, forward: boolean): Optional<() => void> => {
   const fromPos = CaretPosition.fromRangeStart(editor.selection.getRng());
   return CaretFinder.fromPosition(forward, editor.getBody(), fromPos)
     .filter((pos) => forward ? isBeforeImageBlock(pos) : isAfterImageBlock(pos))
     .bind((pos) => Optional.from(getChildNodeAtRelativeOffset(forward ? 0 : -1, pos)))
-    .exists((elm) => {
-      editor.selection.select(elm);
-      return true;
-    });
+    .map((elm) =>
+      () => {
+        editor.selection.select(elm);
+        return true;
```

Comment:
My guess is that you can return things, and it's still valid, it's just that calling functions know not to expect any value? Same way you can call a function with more parameters than it can accept.